### PR TITLE
Support "dummy" relayer type

### PR DESCRIPTION
**Pull request by samsondav**
This allows testing of plugins without needing a real connection to any chain.
Ticket: https://smartcontract-it.atlassian.net/browse/MERC-5574

### CODE REVIEW

File: core/services/relay/relay.go
Comment (jmank88) [inline core/services/relay/relay.go:1]: Does this set still need to be updated? The unmarshaller doesn't require the value to be in this set anymore.

File: core/services/relay/dummy/mercury_provider.go
Comment (jmank88) [inline core/services/relay/dummy/mercury_provider.go:1]: Remove?

File: core/services/relay/dummy/llo_provider.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+func (p *lloProvider) Ready() error {
+	return errors.Join(p.cp.Ready(), p.transmitter.Ready(), p.channelDefinitionCache.Ready())
+}
+
+func (p *lloProvider) Name() string {
+	return p.logger.Name()
+}
+
+func (p *lloProvider) HealthReport() map[string]error {
+	report := map[string]error{}
```
Comment (jmank88) [inline core/services/relay/dummy/llo_provider.go]: Needs to include itself:
```suggestion
	report := map[string]error{hb.Name(): hb.Healthy()}
```

File: core/services/relay/relay.go
Comment (samsondav) [reply core/services/relay/relay.go:1]: Yeah apparently so, there was a test failing if I didn't add this

File: core/services/relay/dummy/llo_provider.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+func (p *lloProvider) OffchainConfigDigester() ocrtypes.OffchainConfigDigester {
+	return p.cp.OffchainConfigDigester()
+}
+
+func (p *lloProvider) OnchainConfigCodec() datastreamsllo.OnchainConfigCodec {
+	// TODO: This should probably be moved to core since its chain-specific
+	// https://smartcontract-it.atlassian.net/browse/MERC-3661
+	return &datastreamsllo.JSONOnchainConfigCodec{}
```
Comment (prashantkumar1982) [inline core/services/relay/dummy/llo_provider.go]: Is it ideal for the dummy relayer to have a dependency on the chainlink-data-streams package?
Would it make sense to use something product agnostic?
Or is this not a big issue to try fixing.

File: core/services/ocr2/delegate.go [hunk 3]
```diff
@@ -895,9 +895,6 @@ func (d *Delegate) newServicesLLO(
 	if err != nil {
 		return nil, ErrJobSpecNoRelayer{Err: err, PluginName: "streams"}
 	}
-	if rid.Network != relay.NetworkEVM {
```
Comment (austinborn) [inline core/services/ocr2/delegate.go:898]: Just to double-check, you don't need this log anymore?

**Review by austinborn — DISMISSED** [review]

File: core/services/ocr2/delegate.go (hunk 3 above)
Comment (samsondav) [reply core/services/ocr2/delegate.go:898]: This was a check, not a log if I'm not mistaken, and there isn't really any reason for it to be here. LLO should support any relayer type.

File: core/services/relay/dummy/llo_provider.go (hunk 2 above)
Comment (samsondav) [reply core/services/relay/dummy/llo_provider.go]: There is a TODO there which probably should be implemented. I'll move it to core :+1:

**Review by krehermann — APPROVED** [review]

**Review by austinborn — APPROVED** [review]

**Pull request merged**